### PR TITLE
TDD  PROJECT

### PR DESCRIPTION
**Pull request by AACHOURMOHAMED**
# In this pull request
> Add Test to solver
> Add factorial methods
> Add reverse method
> Add fizzbuzz method

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  end
+  it 'should return 6 for 3' do
+    expect(Solver.factorial(3)).to eq(6)
+  end
+  it 'should return 24 for 4' do
+    expect(Solver.factorial(4)).to eq(24)
+  end
+  it 'should return 120 for 5' do
+    expect(Solver.factorial(5)).to eq(120)
+  end
```
Comment (GzimAsani) [inline spec/solver_spec.rb:21]: - [ ] The #factorial method need to have tests that handle negative value which raises an exception as an expectation
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @AACHOURMOHAMED and @hammaazarok          :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, they almost covered everything:+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  it 'should return 1 for -1' do
+    expect(Solver.factorial(-1)).to eq(1)
+  end
+  it 'should return 1 for -2' do
+    expect(Solver.factorial(-2)).to eq(1)
+  end
+  it 'should return 1 for -3' do
+    expect(Solver.factorial(-3)).to eq(1)
+  end
```
Comment (rotimiazeez) [inline spec/solver_spec.rb]: - According to the reviewer's suggestion, kindly make sure that when a negative integer is given, it should raise an exception 😉. For example:
```
     describe Solver do
         describe "#factorial" do
             it "raises exception for negative number" do
             solver = Solver.new
             expect { solver.factorial(-5) }.to raise_exception
             end
         end
     end
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi Team 👋🏽, 
### Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/solver_spec.rb (hunk 1 above)
Comment (danmainah) [reply spec/solver_spec.rb:21]: - [ ] @team great work implementing the changes requested.

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
